Support putting run configs in folders

SquidDev opened the pull request:
- Remove unused `RunConfig.genRuns`.
 - Rewrite templating to do a single find-and-replace, rather than a chain of them.
 - Add a new `ideConfigFolder` option, which sets the `folderName` attribute on the IDEA config file.

![IntellJ Idea's run config drop down. There is a folder called "Fabric", where which contains a list of runs (e.g. client, server).](https://github.com/user-attachments/assets/671077df-3402-4943-9608-b82d81d8e7cc)


I wasn't quite sure the best way to expose settings from `RunConfigSettings`, as it's currently a bit ad-hoc. For now, I've gone with the simplest option (a `Property<String>` getter), but happy to change that if preferred!

SquidDev committed:
Remove unused RunConfig.genRuns

SquidDev committed:
Support putting run configs in folders

SquidDev committed:
Fix checkstyle complaints

Given how many of my projects use checkstyle, you'd have thought I'd
remember to check it before submitting a PR.

modmuss50 commented:
Looks like https://github.com/FabricMC/fabric-loom/blob/dev/1.10/src/test/groovy/net/fabricmc/loom/test/unit/IdeaClasspathModificationsTest.groovy is now failing

SquidDev committed:
Fix IdeaClasspathModificationsTest

"I won't run the tests, I've tested manually, and nothing will break".
Words spoken before disaster (or at least making a fool of myself in
CI).

SquidDev commented:
I did have a think about replacing `IdeaClasspathModificationsTest.fromDummy` with a call to `RunConfig.fromDummy`. *Technically* we can get away passing a null `Project` to it, but it felt a bit risky, so have left as-is for now.

modmuss50 commented:
> I did have a think about replacing `IdeaClasspathModificationsTest.fromDummy` with a call to `RunConfig.fromDummy`. _Technically_ we can get away passing a null `Project` to it, but it felt a bit risky, so have left as-is for now.

I'd leave it for now, the test coverage (and testability) of all of this could be improved vastly, but there is no point doing it as part of this PR. Ill take a closer look at the PR shortly and get it merged into the 1.11 branch.
